cmd/process-agent/collector.go

```diff
@@ -235,14 +217,78 @@ func (l *Collector) run(exit chan bool) error {
 					}
 				}
 			}
-		}(c)
+		}(c, payloads)
 	}
+
 	<-exit
-	l.forwarder.Stop()
-	l.podForwarder.Stop()
+	wg.Wait()
+
+	processForwarder.Stop()
+	podForwarder.Stop()
 	return nil
 }
 
+func (l *Collector) consumePayloads(payloads chan checkPayload, fwd forwarder.Forwarder, exit chan bool) {
+	for {
+		select {
+		case payload := <-payloads:
+			if len(payloads) >= l.cfg.QueueSize {
+				log.Info("Expiring payload from in-memory queue.")
```

Comment:
nit: should log what queue is expiring.

Reply:
This is going away in a follow up PR.

I'm sorry that I neglected to mention that this PR was sort of transitional.  I broke up the logic for managing the retry queues in to 2 separate PRs.  This is the first.